src/VecSim/algorithms/hnsw/hnsw_factory.cpp

```diff
@@ -73,3 +74,12 @@ size_t HNSWFactory::EstimateElementSize(const HNSWParams *params) {
      */
     return size_meta_data + size_total_data_per_element;
 }
+
+// TODO overload for doubles
```

Comment:
The whole purpose of the factory is to be a proxy class and avoid the include loop the templates cause.
(since batch iterator and hnsw are templated at least one of them shouldn't include the other's header, so we cant move this function to the factory header..)